src/fixate/_switching.py

```diff
@@ -87,14 +93,73 @@ def __or__(self, other: PinUpdate) -> PinUpdate:
 
 PinUpdateCallback = Callable[[PinUpdate, bool], None]
 
+S = TypeVar("S", bound=str)
 
-class VirtualMux:
+
+class VirtualMux(Generic[S]):
+    map_tree: Optional[TreeDef] = None
+    map_list: Optional[Sequence[Sequence[str]]] = None
     pin_list: PinList = ()
     clearing_time: float = 0.0
 
+    def __class_getitem__(cls, arg):
+        # https://peps.python.org/pep-0560
+
+        # so the problem we have to solve is the way this works is convoluted and the __orig_bases__ property we rely on is not preserved
+        # __orig_bases__ is an attribute set by the metaclass that is used to store the type information provided by __class_getitem__
+        # expected class creation:
+        # VirtualMux[type] -> VirtualMux.__class_getitem__(type) -> VirtualMux.__init__() (with type info visible on __orig_bases__)
+        # what actually happens
+        # VirtualMux[type] -> VirtualMux.__class_getitem__(type) -> _GenericAlias(VirtualMux).__call__() -> VirtualMux.__init__()
+        # the _GenericAlias middle class means we lose the type information provided through the __class_getitem__ call
+
+        # the attribute __orig_class__ can be used to store the original class that was created with __class_getitem__
+        # HOWEVER this relies on the __init__ of VirtualMux succeeding - we end up in a circular dependency of needing
+        # the init to succeed to set the attribute, but needing the attribute to be set to succeed in the init (due to our implementation)
```

Comment:
General code organisation point. Can you please move this down to the bottom of the class definition where the other internal details are?

Also, please wrap the comments to ~100 columns.

Reply:
I figured out how to get a ruler in vscode so will happen way less often now if I forget to run my formatter.